Theming med variabler

antidecaf opened the pull request:
Oppdatert av @kwltrs: Tilrettelegging for theming av ffe ved hjelp av sentraliserte variabler (_css custom properties_).

Pakkene `ffe-core`, `ffe-buttons` og `ffe-form` kan "themes" i det man overstyrer css custom properties. Sentrale properties defineres i `ffe-core/less/theme.less` med navn `--ffe-g-*`, og hver pakke har en tilsvarende `theme.less` med lokale properties `--ffe-v-*`.

I samme slengen fjernes de less-mixins for font-størrelser (`.ffe-fontsize-h1()` osv.).

antidecaf committed:
feat(ffe-buttons): theming med custom properties

BREAKING CHANGE: introduces css custom properties for colors.

kwltrs commented:
Bygget blir grønt om du rebaser på siste endringer på develop-branchen.

kwltrs committed:
feat(component-overview): legger til et eksempel av theming

kwltrs commented:
I commit f840a448b2a647502819d623a71832cbb7fca0e1 er et eksempel hvordan themingen kan se ut.

kwltrs committed:
feat(ffe-core): theming med custom properties

BREAKING CHANGE: introduces css custom properties for colors and fonts.

kwltrs committed:
feat(ffe-form): theming med custom properties

BREAKING CHANGE: introduces css custom properties for colors.

BREAKING CHANGE: removes the less mixin `ffe-sb1-radioblob()`.

kwltrs committed:
feat(component-overview): use css properties from less, not css

kwltrs committed:
feat(ffe-core): removes .ffe-icon-base

BREAKING CHANGE: removes outdated css class .ffe-icon-base.

kwltrs committed:
feat(ffe-core): add custom props for font sizes

BREAKING CHANGE: this change removes the less mixins `ffe-fontsize-*`.

kwltrs committed:
feat(ffe-buttons): use custom props for font sizes

kwltrs committed:
feat(ffe-form): use custom props for font sizes

kwltrs committed:
feat(ffe-lists): use custom props for font sizes

kwltrs committed:
refactor(ffe-cards): remove unused code

HeleneKassandra commented on packages/ffe-cards/less/components.less:
```diff
@@ -22,9 +22,6 @@
     &__title {
         &:extend(.ffe-h5);
-        /* stylelint-disable-next-line block-no-empty */
-        .native &:extend(.native .ffe-h5) {
```
Fjernes denne fordi det blir dobbelt opp? eller hva skjer her?

kwltrs replied:
Den kan fjernes fordi det ikke lengre finnes en regel for `.native .ffe-h5`. ~Reglen for `.native .ffe-h5` kom fra `ffe-core/less/typography.less` og ble generert med [.ffe-fontsize-h5() ](https://github.com/SpareBank1/designsystem/blob/9955a2b5fc44a9ec0481f59a4bce152c3c6f3dc7/packages/ffe-core/less/font-sizes.less#L60). Men det er nå løst med å endre variablen `--ffe-fontsize-h5` i [en media query](https://github.com/SpareBank1/designsystem/blob/e7aa6ebc249054ac322aeaff8126b9c599d0a064/packages/ffe-core/less/theme.less#L85).~
Stemmer ikke. Reglen for `.native .ffe-h5` var i [ffe-core/less/typography.less](https://github.com/SpareBank1/designsystem/blob/9955a2b5fc44a9ec0481f59a4bce152c3c6f3dc7/packages/ffe-core/less/typography.less#L210) men erstattes av en [annen media query](https://github.com/SpareBank1/designsystem/blob/e7aa6ebc249054ac322aeaff8126b9c599d0a064/packages/ffe-core/less/theme.less#L154) enn jeg skrev.

HeleneKassandra commented on packages/ffe-form/less/checkbox.less:
```diff
@@ -56,13 +54,7 @@
-                background-color: @ffe-farge-svart;
-            }
-        }
+        background-color: var(--ffe-v-input-bg-color);
```
Litt usikker på hvor nøyaktig, men darkmode fargene på checkbox har blitt endret i denne branchen. Så nå er den lik som i lightmode.

HeleneKassandra replied:
Ser ut som fargene er riktig men teksten til checkbox blir nå ikke sentrert med selve checkboxen?  Nå blir teksten plassert ved bunnen.  
![Skjermbilde 2022-10-19 kl  12 24 58](https://user-images.githubusercontent.com/39946146/196665959-e23b6169-d35f-4071-828e-f4238a660e09.png)

HeleneKassandra commented on packages/ffe-form/less/form-label.less:
```diff
@@ -12,20 +12,14 @@
-            color: @ffe-farge-vann-70;
-        }
-    }
+    color: var(--ffe-g-secondary-color);
```
Her endrer du label fargen i darkmode, Den blir lysere enn det den pleide å være. LItt usikker på om det er med vilje eller ikke?

kwltrs replied:
Det burde være som i figma. Fjell i light mode, vann-70 i dark mode.

HeleneKassandra commented on packages/ffe-form/less/input-field.less:
```diff
@@ -16,15 +16,15 @@
+    background-color: var(--ffe-v-input-bg-color);
+    color: var(--ffe-v-input-color);
+    border-radius: var(--ffe-g-border-radius);
+    border: 2px solid var(--ffe-g-border-color);
```
Border color på input field og dropdown etc er lysere nå enn de var før endringen i lightmode

kwltrs replied:
Ja, det ble feil grå-farge. Burde ha vært varm grå og ikke lys varmgrå.

HeleneKassandra commented on packages/ffe-form/less/radio-switch.less:
```diff
@@ -17,13 +17,13 @@
-    color: @ffe-farge-vann;
-    font-family: 'SpareBank1-medium', 'MuseoSansRounded-700', arial, sans-serif;
+    color: var(--ffe-g-primary-color);
+    font-family: var(--ffe-g-font);
```
Her tror jeg du erstatter med en mindre bold tekst enn det som er der i dag

antidecaf replied:
Stemmer, denne bør vel være `--ffe-g-font-heading-small` eller `--ffe-g-font-strong`

HeleneKassandra replied:
![Skjermbilde 2022-10-19 kl  12 27 01](https://user-images.githubusercontent.com/39946146/196666405-fecbd140-e361-49ed-ae6b-528b65dbfda4.png)
Fonten er fikset, men vi får nå en bakgrunnsfarge. Den grå sirkelen skal være sort eller transparent i forhold til sånn den er i dag

HeleneKassandra commented on packages/ffe-form/less/radio-switch.less:
```diff
@@ -17,13 +17,13 @@
     position: relative;
     background-color: @ffe-farge-hvit;
-    border: 2px solid @ffe-farge-varmgraa;
+    border: 2px solid var(--ffe-g-border-color);
```
Borderen blir lysere enn den er før endring

HeleneKassandra commented on packages/ffe-form/less/toggle-switch.less:
```diff
@@ -67,19 +67,13 @@
             width: 56px;
             height: 30px;
-            background: @ffe-farge-varmgraa;
+            background: var(--ffe-g-border-color);
```
Her blir bakgrunnen lysere enn den var i lightmode

antidecaf commented on packages/ffe-form/less/theme.less:
```diff
@@ -0,0 +1,22 @@
+    --ffe-v-error-message-icon-color: var(--ffe-g-error-color);
+    --ffe-v-checkbox-color: var(--ffe-g-border-color);
+    --ffe-v-checkbox-selected-color: var(--ffe-g-primary-color);
+    --ffe-v-checkbox-corona-color: var(--ffe-g-primary-color);
```
... Corona? 😄

antidecaf replied:
Helt alvorlig, jeg skjønner ikke helt hva dette variabelnavnet betyr?

kwltrs replied:
Det er ringen rundt checkboxer når de har fokus. Korona betyr i dette tilfellet lyskrans -> https://no.wikipedia.org/wiki/Korona
![image](https://user-images.githubusercontent.com/493924/196635949-c70b0f62-e855-4fe4-a0b8-5d9cb864baef.png)

HeleneKassandra replied:
Er det bedre å rename det til focus eller focus ring?

HeleneKassandra commented on packages/ffe-form/less/tooltip.less:
```diff
@@ -13,68 +13,45 @@
         border-radius: 50%;
-        border: 2px solid @ffe-farge-varmgraa;
-        color: @ffe-farge-vann;
+        border: 2px solid var(--ffe-v-checkbox-color);
```
Border blir lysere enn den er før endring i lightmode. Får basically darkmode fargen i lightmode også

HeleneKassandra replied:
Det er en minimal forskjell mellom border i darkmode etter endring enn før.  Den er hakket lysere på versjonen før endringen. Lightmode er nå fikset :)

HeleneKassandra reviewed (changes requested):
Noen komponenter får noen farge endringer som leder til for dårlig kontrast. Mistenker at det skjer fordi man fjerner darkmode stylingen og setter darkmode fargen på lightmode også?

HeleneKassandra reviewed (approved):
Supert :D
